server/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -1021,10 +987,13 @@ private BytesReference buildMessage(long requestId, byte status, Version nodeVer
      */
     public void inboundMessage(TcpChannel channel, BytesReference message) {
         try {
+            channel.getStats().markRead();
```

Comment:
I tend to think that a few more things will eventually end up on `ChannelStats`. I'm trying to avoid continued growth of the `TcpChannel` interface as all implementations must implement it. I would prefer at this moment to just leave the logic in `ChannelStats` so that each `TcpChannel` implementation only has to implement the logic of returning the instance.

Reply:
fair enough. that is a good reason.